src/decorators/bind.ts

```diff
@@ -171,3 +171,16 @@ function createTypeGuard(binding: TwoWayBinding) {
     return binding.typeGuard ? binding.typeGuard(value) : true;
   };
 }
+
+function scheduleIsComponentCheck(binding: BindSuperConfig) {
+  setTimeout(() => {
```

Comment:
Execution order. @component is called after @bind, so @bind can't know at that point whether it was attached to a component. And since @component takes care of initializing the binding there is no other point in time where @bind can perform such a check.

Reply:
Why not perform this check in `@component` instead?

Reply:
... because this check is there to verify that @component is called? That's like planning to checking your shoes are tied only whenever you tie them.

Reply:
Oh, I see, this is only a measure to ensure that `@component` is applied, but `@bind` does not require anything that `@component` offers.